_posts/2020-03-16-dask-usage-at-blue-yonder.markdown

```diff
@@ -0,0 +1,161 @@
+---
+layout: single
+title:  "Dask Usage at Blue Yonder"
+date:   2020-03-16 10:00:00 +0100
+tags: technology python data-engineering
+header:
+  overlay_image: assets/images/tech_gear_banner.jpg
+  overlay_filter: 0.2
+  show_overlay_excerpt: false
+author: Andreas Merkel, Kshitij Mathur
+author_profile: true
+---
+# Dask Usage at Blue Yonder
+
+Back in 2017, Blue Yonder started to look into 
+[Dask/Distributed](https://distributed.dask.org) and how we can leverage it to power
+our machine learning and data pipelines.
+It's 2020 now, and we are using it heavily in production for providing demand predictions,
+recommended order quantities, and price proposals to our customers.
+Time to take a a look at the way we are using Dask!
+
+## Use cases
+
+First, let's have a look at the use cases we have for Dask. The use of Dask at Blue Yonder
+is strongly coupled to the concept of [Datasets](https://tech.jda.com/introducing-kartothek/),
+tabular data stored as Apache [Parquet](https://parquet.apache.org/) files in blob storage. 
+We use these datasets for storing everything that is needed
+for generating predictions, recommended order quantities, and recommended prices for our customers,
+as well as the predictions, order quantities, and prices themselves.
+
+The size of the data we work on varies strongly depending on the individual customer, with the largest ones
+currently amounting to about 300 GiB of compressed Parquet data per day, corresponding to about 500 GiB of
+uncompressed data in memory.
+
+### Downloading data from a database to a dataset
+
+One typical use case that we have is downloading large amounts of data from a database into a
+dataset. We use Dask/Distributed here as a means to parallelize the download and the graph is
+an extremely simple one: a lot of parallel nodes downloading pieces of data into a blob and one
```

Comment:
The technical term for "extremely simple" graph in this case would be [embarrassingly parallel](https://en.wikipedia.org/wiki/Embarrassingly_parallel)